docs/scenario-controller.md

```diff
@@ -0,0 +1,43 @@
+<div align="center"><h1>IntegrationTestScenario Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>Monitor IntegratonTestScenario <br>& filter only created/updated <br>events for the resource ))
+  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureCreatedScenarioIsValid() function
+
+  %% Node definitions
+  
+  application_exists{"Application for scenario <br>was found?"}
+  set_owner_reference(Set owner reference to <br>IntegrationTestScenario <br> if not already existing)
+  environment_defined{"IntegrationTestScenario <br>has environment defined?"}
+  environment_exists{"Environment exists <br> in same namespace <br> as IntegrationTestScenario?"}
+  update_scenario_status_valid(Update IntegrationTestScenario <br>status to valid)
+  update_scenario_status_invalid(Update IntegrationTestScenario <br>status to invalid)
+  complete_reconciliation(Complete reconciliation for <br>IntegrationTestScenario)
+  continue_reconciliation(Continue with next reconciliation)
+
+
+  %% Node connections
+  predicate                        ---->    |"EnsureCreatedScenarioIsValid()"| application_exists
+  application_exists               --No-->  continue_reconciliation
```

Comment:
In case, the application is not found, we first mark the scenario as "Invalid" before going to the next reconciliation loop. Please see:
https://github.com/redhat-appstudio/integration-service/blob/e77373aac7d358fd2b4eb30c5f870f98373203aa/controllers/scenario/scenario_adapter.go#L63